src/server.ts

```diff
@@ -0,0 +1,43 @@
+import {
+	addTodo,
+	removeTodo,
+	listAllTodos,
+	markTodoAsDone,
+} from "./todo_final";
+
+// for req.body
+const bodyParser = require("body-parser");
+
+// require the just installed express app
+const express = require("express");
+const path = require("path");
+// then we call express
+const app = express();
+// json express convention
+app.use(express.json());
+
+// takes us to the root(/) URL. returns list of todos in JSON format and appropriate HTTP status code.
+app.get("/", function (req: any, res: any) {
+	res.json(listAllTodos());
+});
+
+// - Post @ `/` - adds a todo. Returns the newly added object and appropriate HTTP status code.
+app.post("/", function (req: any, res: any) {
+	res.json(addTodo(req.body.text, req.body.dueAtstr));
+});
+
+// Put @ /{id}/done - marks a todo as done and returns a HTTP status code to indicate success/failure.
+app.put("/:id/done", function (req: any, res: any) {
+	console.log(typeof req.params.id);
+	res.send(markTodoAsDone(parseInt(req.params.id)));
+});
+
+// Delete @ /{id} - delete a todo item. Returns the deleted object and the appropriate HTTP status code.
+app.delete("/:id", function (req: any, res: any) {
+	res.send(removeTodo(req.params.id));
+});
```

Comment:
Three things:
* Testing with 3 items reveals that this endpoint always removes the last todoItem regardless of the `id` you pass in. This is beceause `removeTodo` is designed to take a `TodoItem` as input, not an `id`, and you are passing it an `id`. I think you will need to add a `findTodoById` function that works similarly to `findTodoByText`, then use the new function to make this endpoint work. You will have to use `parseInt`

* This endpoint is supposed to return the deleted todoItem. You will need to modify `removeTodo` so that it returns the todoItem it deletes. You will probably need to return the todoItem from the endpoint in JSON format.

* This endpoint needs to return an error status code if an item with a matching id wasn't found.

Reply:
How do you know something needs to be return in JSON format